script.ts

```diff
@@ -0,0 +1,247 @@
+var displ = <HTMLInputElement>document.getElementById("display");
+
+function evalu(): void {
+    displ.value = eval(displ.value).toFixed(2);
+}
+
+function deleteChar(): void {
+    displ.value = displ.value.substring(0, displ.value.length - 1)
+}
+
+function fe(): void {
+    displ.value = Number(displ.value).toExponential();
+}
+
+function exp(): void {
+    displ.value = String(Math.exp(Number(displ.value)));
+}
+
+function fact(x)
+  {
+if (x === 0)
+ {
+    return 1;
+ }
+ return x * fact(x-1);
+}
+
+function factorial(x) 
```

Comment:
same comment as above:
what type this function will return? add type for it
here you are taking x as argument what is the type for x is also missing